packages/perseus/src/util/graphie-utils.ts

```diff
@@ -0,0 +1,253 @@
+import {Errors, PerseusError} from "@khanacademy/perseus-core";
+
+import {getDependencies} from "../dependencies";
+import {Log} from "../logging/log";
+import Util from "../util";
+
+import type {Coord} from "../interactive2/types";
+import type {CSSProperties} from "aphrodite";
+
+// For offline exercises in the mobile app, we download the graphie data
+// (svgs and localized data files) and serve them from the local file
+// system (with file://). We replace urls that start with `web+graphie`
+// in the perseus json with this `file+graphie` prefix to indicate that
+// they should have the `file://` protocol instead of `https://`.
+const svgLocalLabelsRegex = /^file\+graphie:/;
+const hashRegex = /\/([^/]+)$/;
+
+/**
```

Comment:
These types needed to come along to support the parsing. I'm exporting them for Graphie2000, but I do wonder which side should be in charge of these.

Reply:
I think it makes sense for Perseus to own them. They are a part of how `SvgImage` works and consumes graphie images.